src/main/java/java_echo_server/App.java

```diff
@@ -1,57 +1,22 @@
 package java_echo_server;
 
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.PrintWriter;
 import java.net.ServerSocket;
-import java.net.Socket;
 
 public class App {
-    private static ServerSocket serverSocket;
-    private static Socket clientSocket;
-
-    public static void main(String[] args) {
-        int portNumber = setPortNumber(args);
-        Client client = createClientSocketWrapper(portNumber);
-
-        EchoServer echoServer = new EchoServer(client);
-
-        do {
-            echoServer.start();
-        } while (echoServer.isRunning());
-        stopServer();
-    }
-
-    private static int setPortNumber(String[] terminal_args) {
-        return terminal_args.length > 0 ? Integer.parseInt(terminal_args[0]) : 1234;
-    }
-
-    private static Client createClientSocketWrapper(int port) {
-        try {
-            serverSocket = new ServerSocket(port);
-            clientSocket = serverSocket.accept();
-
-            Boolean autoFlushWriter = true;
-            PrintWriter printWriter = new PrintWriter(clientSocket.getOutputStream(), autoFlushWriter);
-            WriterWrapper outputStream = new PrintWriterWrapper(printWriter);
-            BufferedReader inputStream = new BufferedReader((new InputStreamReader(clientSocket.getInputStream())));
-
-            return new Client(inputStream, outputStream);
-
-        } catch (IOException e) {
-            System.err.println(e);
-            return null;
+        public static void main(String args[]) {
+            try {
+                ServerSocket serverSocket = new ServerSocket(setPortNumber(args));
+                Thread serverThread = new Thread(new EchoProtocol(serverSocket), "SERVER");
+                serverThread.start();
+            } catch (IOException e) {
+                System.err.println(e.toString());
+            }
         }
-    }
 
-    private static void stopServer() {
-        try {
-            System.out.println("The server is shutting down.");
-            clientSocket.close();
-            serverSocket.close();
-        } catch (IOException e) {
-            System.err.println("The server could not be shut down.");
+        private static int setPortNumber(String[] terminal_args) {
```

Comment:
more spacing here than elsewhere—why?

Reply:
I tabbed the ternary condition so that it would be read under terminal_args and not return.
<img width="512" alt="Screen Shot 2019-06-06 at 10 37 35 AM" src="https://user-images.githubusercontent.com/29721784/59053817-1ea9a280-8847-11e9-852e-a57ebe9b2ec2.png">
<img width="496" alt="Screen Shot 2019-06-06 at 10 37 40 AM" src="https://user-images.githubusercontent.com/29721784/59053823-21a49300-8847-11e9-91d6-0435be5b095c.png">

But this is a very trivial reason and I'll adjust it back.

Reply:
Sorry, I meant the entire file seemed like it had more spacing

Reply:
Ahhhh, Ok -- Adjusting!